src/js/client/hasRequiredOptions.js

```diff
@@ -0,0 +1,41 @@
+// Grade to add checking for required options on startup.
+// TODO: Migrate to using JSON Schema validation to handle this once this is resolved: https://issues.gpii.net/browse/GPII-1176
+/* global jQuery, fluid */
+(function () {
+    "use strict";
+    var gpii = fluid.registerNamespace("gpii");
+    fluid.registerNamespace("gpii.templates.templateAware");
+
+    // A static function to check for the existing of required options data and fail, typically called on startup.
+    //
+    // `that.options.requiredFields` is expected to be an object whose keys represent relative paths to definitions, as in:
+    //
+    // requiredFields: {
+    //   "path.relative.to.that.options": true
+    // }
+    //
+    gpii.checkRequiredOptions = function (that, location) {
```

Comment:
Repackage this slightly so that it can be used for non-components and other cases - I suggest a signature

gpii.checkRequiredOptions = function (options, requiredFields, location, suffix) {

and then the listener below can be adjusted to read:

args: ["{that}.options", "{that}.options.requiredFields", location, "component"]